src/de/mangatube/src/eu/kanade/tachiyomi/extension/de/mangatube/MangaTube.kt

```diff
@@ -40,146 +35,225 @@ class MangaTube : ParsedHttpSource() {
+    override fun popularMangaParse(response: Response): MangasPage {
+        val body = MangaTubeHelper.checkResponse(response)
 
-    override fun popularMangaFromElement(element: Element): SManga = throw UnsupportedOperationException()
+        val res: BaseResponse<MangasWrapper> = json.decodeFromString(body)
 
-    override fun popularMangaNextPageSelector() = throw UnsupportedOperationException()
+        if(!res.success){
+            throw Exception("Something went wrong!")
+        }
+
+
+        val mangaList = res.data.manga.map { manga ->
+            mangas[manga.title] = manga
+            SManga.create().apply {
+                title = manga.title
+                url = "$baseUrl${manga.url}"
+                thumbnail_url = manga.cover
+                genre = manga.genre.map { genre -> Genre.fromId(genre)!! }
+                    .joinToString(", ") { genre -> genre.displayName }
+                status = MangaTubeHelper.mangaStatus(manga.status)
+            }
+        }
 
-    // Latest
+        return MangasPage(mangaList, false)
+    }
 
     override fun latestUpdatesRequest(page: Int): Request {
-        return GET("$baseUrl/?page=$page", headers)
+        val url = "$baseUrl/api/home/new-manga"
+
+        Log.d("MangaTube", "Request new mangas")
+        Log.d("MangaTube", "Url -> $url")
+
+        return GET(url)
     }
 
-    override fun latestUpdatesSelector() = "div#series-updates div.series-update:not([style\$=none])"
+    override fun latestUpdatesParse(response: Response): MangasPage {
+        val body = MangaTubeHelper.checkResponse(response)
 
-    override fun latestUpdatesFromElement(element: Element): SManga {
-        return SManga.create().apply {
-            element.select("a.series-name").let {
-                title = it.text()
-                setUrlWithoutDomain(it.attr("href"))
+        val res: BaseResponse<MangasWrapper> = json.decodeFromString(body)
+
+        if(!res.success){
+            throw Exception("Something went wrong!")
+        }
+
+
+        val mangaList = res.data.manga.map { manga ->
+            mangas[manga.title] = manga
+            SManga.create().apply {
+                title = manga.title
+                url = "$baseUrl${manga.url}"
+                thumbnail_url = manga.cover
+                genre = manga.genre.map { genre -> Genre.fromId(genre)!! }
+                    .joinToString(", ") { genre -> genre.displayName }
+                status = MangaTubeHelper.mangaStatus(manga.status)
+                description = manga.description
             }
-            thumbnail_url = element.select("div.cover img").attr("abs:data-original")
         }
+
+        return MangasPage(mangaList, false)
     }
 
-    override fun latestUpdatesNextPageSelector() = "button#load-more-updates"
+    override fun mangaDetailsRequest(manga: SManga): Request {
+        val url = "$baseUrl/api/manga/${mangas[manga.title]!!.id}"
```

Comment:
save the id in the actual SManga url when returning, the hashmap above will not be initialized on app restart